when i try make, i get the error, thx for your reply

[root@iZbp17euuofuecjbal4r90Z xgboost]# make
Makefile:31: MAKE [make] - checked OK
g++ -DDMLC_LOG_CUSTOMIZE=1 -std=c++11 -Wall -Wno-unknown-pragmas -Iinclude   -Idmlc-core/include -Irabit/include -I/include -O3 -funroll-loops -fPIC -DDISABLE_OPENMP -MM -MT build/tree/tree_model.o src/tree/tree_model.cc >build/tree/tree_model.d
g++ -c -DDMLC_LOG_CUSTOMIZE=1 -std=c++11 -Wall -Wno-unknown-pragmas -Iinclude   -Idmlc-core/include -Irabit/include -I/include -O3 -funroll-loops -fPIC -DDISABLE_OPENMP src/tree/tree_model.cc -o build/tree/tree_model.o
src/tree/tree_model.cc: In constructor ‘xgboost::GraphvizGenerator::GraphvizGenerator(const xgboost::FeatureMap&, const string&, bool)’:
src/tree/tree_model.cc:465:55: error: invalid initialization of non-const reference of type ‘std::stringstream& {aka std::basic_stringstream<char>&}’ from an rvalue of type ‘<brace-enclosed initializer list>’
       TreeGenerator(fmap, with_stats), ss_{SuperT::ss_} {
                                                       ^
make: *** [build/tree/tree_model.o] Error 1

--- a/src/tree/tree_model.cc
+++ b/src/tree/tree_model.cc
@@ -457,12 +457,11 @@ DMLC_REGISTER_PARAMETER(GraphvizParam);
 
 class GraphvizGenerator : public TreeGenerator {
   using SuperT = TreeGenerator;
-  std::stringstream& ss_;
   GraphvizParam param_;
 
  public:
   GraphvizGenerator(FeatureMap const& fmap, std::string const& attrs, bool with_stats) :
-      TreeGenerator(fmap, with_stats), ss_{SuperT::ss_} {
+      TreeGenerator(fmap, with_stats) {
     param_.InitAllowUnknown(std::map<std::string, std::string>{});
     using KwArg = std::map<std::string, std::map<std::string, std::string>>;
     KwArg kwargs;

